lib/core/address_validator.dart

```diff
@@ -336,7 +336,9 @@ class AddressValidator extends TextValidator {
     }
 
     if (pattern != null) {
-      return "$BEFORE_REGEX($pattern)$AFTER_REGEX";
+      if (requireSurroundingWhitespaces)
+        return "$BEFORE_REGEX($pattern)$AFTER_REGEX";
```

Comment:
I believe this regex should already handle whether it is preceded by space or not, so this should work as well, if not then the regex is wrong, can you check that instead

Reply:
If that check would be optional, what would be the point of having it in the regex at all?